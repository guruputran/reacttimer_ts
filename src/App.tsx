import React, { Component } from 'react'; 
import "bootstrap/dist/css/bootstrap.css"; // Import precompiled Bootstrap css
import styled from 'styled-components';

type ClockState = {
  time: Date
}

const StyledDiv = styled.div`
font-family: Roboto-Bold;
font-size: 84px;
color: #37474F;
display: flex;
flex-direction: column;
justify-content: center;
align-items: center;
text-align: center;
min-height: 100vh;
`;

export default class App extends Component<{}, ClockState> {

  tick() {
    this.setState({
      time: new Date()
    });
  }

  componentWillMount() {
    this.tick();
  }

  componentDidMount() {
    setInterval(() => this.tick(), 1000);
  }

  render() {
    return <StyledDiv>{this.state.time.toLocaleTimeString()}</StyledDiv>
  }
}