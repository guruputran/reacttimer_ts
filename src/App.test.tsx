import React from 'react';
import { render, screen } from '@testing-library/react';
import App from './App';
import renderer from 'react-test-renderer'

describe("snapshot testing", () => {

  test("snapshot for App component", () => {
    const renderedComponent = renderer.create(<App />).toJSON()
    expect(renderedComponent).toMatchSnapshot()
  })
})
