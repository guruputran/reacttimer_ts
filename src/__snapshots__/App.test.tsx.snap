// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing snapshot for App component 1`] = `
<div
  className="sc-bdnxRM dhuGlk"
>
  1:00:20 am
</div>
`;
